source/simulator/assemblers/stokes.cc

```diff
@@ -134,29 +139,146 @@ namespace aspect
                     {
                       data.local_matrix(i, j) += ((2.0 * eta * (scratch.grads_phi_u[i]
                                                                 * scratch.grads_phi_u[j]))
-                                                  + one_over_eta * pressure_scaling
-                                                  * pressure_scaling
-                                                  * (scratch.phi_p[i]
-                                                     * scratch.phi_p[j]))
+                                                 )
                                                  * JxW;
                     }
 
+
             }
-          else
+          if (this->get_parameters().use_bfbt)
             {
+              const double sqrt_eta=sqrt(eta);
               const unsigned int pressure_component_index = this->introspection().component_indices.pressure;
+
               for (unsigned int i = 0; i < stokes_dofs_per_cell; ++i)
-                if (scratch.dof_component_indices[i] == pressure_component_index)
+                {
                   for (unsigned int j = 0; j < stokes_dofs_per_cell; ++j)
-                    if (scratch.dof_component_indices[j] == pressure_component_index)
-                      {
-                        data.local_matrix(i, j) += (one_over_eta * pressure_scaling
-                                                    * pressure_scaling
-                                                    * (scratch.phi_p[i]
-                                                       * scratch.phi_p[j]))
+                    {
+
+
+                      // i and j are not pressures
+                      if (scratch.dof_component_indices[i]!=pressure_component_index && scratch.dof_component_indices[j]!=pressure_component_index)
+                        data.local_inverse_lumped_mass_matrix[i]+=sqrt_eta*scalar_product(scratch.phi_u[i],scratch.phi_u[j])*JxW;
+
+
+                      // i and j are pressures?
```

Comment:
be consistent about the question mark here and in line 159. Either both need it, or none.